simplq/src/components/pages/Status/QueueDetails.jsx

```diff
@@ -1,31 +1,29 @@
-import React, { useState, useEffect } from 'react';
+import React, { useState, useEffect, useCallback } from 'react';
 import InfoIcon from '@material-ui/icons/Info';
-import QueueStats from '../../common/QueueStats';
-import { QueueRequestFactory } from '../../../api/requestFactory';
+import QueueStats from 'components/common/QueueStats';
+import { useGetQueueStatus } from 'store/queues';
+import { useDispatch } from 'react-redux';
 import SidePanelItem from '../../common/SidePanel/SidePanelItem';
-import useRequest from '../../../api/useRequest';
 
-export default (props) => {
-  const [queueStatusResponse, setQueueStatusResponse] = useState();
-  const { requestMaker } = useRequest();
+export default ({ queueId }) => {
+  const [queueStatus, setQueueStatus] = useState();
+  const dispatch = useDispatch();
+  const getQueueStatus = useCallback(useGetQueueStatus(), []);
 
   useEffect(() => {
-    async function fetchData() {
-      const response = await requestMaker(QueueRequestFactory.getStatus(props.queueId));
-      setQueueStatusResponse(response);
-    }
-    fetchData();
-  }, [props.queueId, requestMaker]);
+    // TODO: Can this state be reused? Do we need a reducer for it?
```

Comment:
Queue status should also be stored in global state? What do you think?

Reply:
We can decide later. It makes sense only if it can be reused from the state. Maybe it can be preserved when going from the Join page to the Status page.

Reply:
For me it's more about uniformity.

Reply:
On second look, it needs to be in the store, and async actions should be split into separate files, possibly in its own folder.